apps/teams-test-app/src/components/DialogAPIs.tsx

```diff
@@ -1,90 +1,125 @@
+      },
+    });
 
-  const resizeDialog = (dialogInfoInput: string): void => {
-    const dialogInfo: DialogInfo = JSON.parse(dialogInfoInput);
-    dialog.resize(dialogInfo);
-    setResizeRes('Teams client SDK call dialog.resize was called');
-  };
+  const SendMessageToChild = (): ReactElement =>
+    ApiWithTextInput<string>({
+      name: 'sendMessageToChild',
+      title: 'sendMessageToChild',
+      onClick: async message => {
+        if (childWindowRef.current && childWindowRef.current !== null) {
+          const childWindow = childWindowRef.current;
+          await childWindow.postMessage(message);
+          return Promise.resolve('Message sent to child');
+        } else {
+          return Promise.resolve('childWindow doesnt exist');
+        }
+      },
+    });
 
-  const submitDialogWithInput = (submitDialogInput: string): void => {
-    if (submitDialogInput.length == 0) {
-      dialog.submit();
-      setSubmitRes('Teams client SDK call dialog.submit was called with no arguments');
-    } else {
-      try {
-        const parsedInput = JSON.parse(submitDialogInput);
-        dialog.submit(parsedInput.result, parsedInput.appIds);
-        setSubmitRes('Teams client SDK call dialog.submit was called with arguments');
-      } catch (error) {
-        if (error instanceof SyntaxError) {
-          setSubmitRes(generateJsonParseErrorMsg());
-        } else if (error instanceof Error) {
-          setSubmitRes(error.message);
+  const SendMessageToParent = (): ReactElement =>
+    ApiWithTextInput<string>({
+      name: 'sendMessageToParent',
+      title: 'sendMessageToParent',
+      onClick: async message => {
+        const parentWindow = ParentAppWindow.Instance;
+        if (parentWindow) {
+          await parentWindow.postMessage(message);
+          return Promise.resolve('Message sent to parent');
         } else {
-          setSubmitRes(JSON.stringify(error));
+          return Promise.resolve('parentWindow doesnt exist');
         }
-      }
-    }
-  };
+      },
+    });
 
-  const checkDialogCapability = (): void => {
-    if (dialog.isSupported()) {
-      setCapabilityCheckRes('Dialog module is supported');
-    } else {
-      setCapabilityCheckRes('Dialog module is not supported');
-    }
-  };
+  const RegisterForParentMessage = (): ReactElement =>
+    ApiWithoutInput({
+      name: 'registerForParentMessage',
+      title: 'registerForParentMessage',
+      onClick: async setResult => {
+        const parentWindow = ParentAppWindow.Instance;
+        parentWindow.addEventListener('message', (message: string) => {
+          setResult(message);
+        });
+        return Promise.resolve('Completed');
+      },
+    });
+
+  const SubmitDialogWithInput = (): ReactElement =>
+    ApiWithTextInput<{ result?: string | object; appIds?: string | string[] }>({
+      name: 'dialogSubmitWithInput',
+      title: 'Dialog Submit With Input',
+      onClick: {
+        validateInput: input => {},
```

Comment:
because both the properties on the type you passed into `ApiWithTextInput<...>` are optional, you should be able to use the other `onClick` version, which just takes a simple delegate:

```
onClick: async submitInput => {
    dialog.submit(submitInput.result, submitInput.appIds);
    return '';
}
```

If that's giving you an error, I'd love to know what that error is, because that means what I intended to work doesn't :)